uefi-test-runner/src/proto/media/mod.rs

```diff
@@ -60,12 +63,35 @@ pub fn test(image: Handle, bt: &BootServices) {
             };
             info!("Root directory entry: {:?}", file_info);
         }
-        directory.reset_entry_readout().unwrap();
+        root_directory.reset_entry_readout().unwrap();
 
-        test_file_system_info(&mut directory);
+        test_file_system_info(&mut root_directory);
+
+        assert_eq!(Ok(true), root_directory.is_directory());
+        assert_eq!(Ok(false), root_directory.is_regular_file());
+
+        info!("creating file in root volume");
+        let created_file = root_directory
+            .open(
+                CString16::try_from("foo").unwrap().as_ref(),
+                FileMode::CreateReadWrite,
+                FileAttribute::empty(),
+            )
+            .unwrap();
+        assert_eq!(Ok(true), created_file.is_regular_file());
+        let created_file = match created_file.into_type().unwrap() {
+            FileType::Regular(file) => file,
+            _ => panic!("unsupported value"),
+        };
+        assert_eq!(Ok(true), created_file.is_regular_file());
+        assert_eq!(Ok(false), created_file.is_directory());
```

Comment:
File creation is already tested in known_disk.rs: https://github.com/rust-osdev/uefi-rs/blob/8f009c0b0dee31aa35a2f68d00c14dac80d80b5c/uefi-test-runner/src/proto/media/known_disk.rs#L127

So I think these `is_regular_file()/is_directory()` checks can be moved there.

Reply:
ohh, thank you! I think, the test name could be a more descriptive 😅 thanks for the pointer!